### PR TITLE
add shortcut to toggle multipage

### PR DESCRIPTION
**Pull request by bakkot**
The shortcut is `m`.

### CODE REVIEW

File: js/menu.js [hunk 1]
```diff
@@ -939,6 +939,37 @@ function makeLinkToId(id) {
...
+        }
+      }
+      location = pathParts.slice(0, -2).join('/') + '/' + hash;
+    } else {
+      location = 'multipage/' + hash;
+    }
+  }
+}
+
+document.addEventListener('keydown', doShortcut);
```
Comment (michaelficarra) [inline js/menu.js:971]: `keypress`?

File: js/menu.js [hunk 2]
```diff
@@ -939,6 +939,37 @@ function makeLinkToId(id) {
...
+  }
+  let target = e.target;
+  let name = target.nodeName.toLowerCase();
+  if (name === 'textarea' || name === 'input' || name === 'select' || target.isContentEditable) {
+    return;
+  }
+  if (e.key === 'm' && !e.altKey && !e.ctrlKey && !e.metaKey && !e.shiftKey && usesMultipage) {
+    let pathParts = location.pathname.split('/');
+    let hash = location.hash;
+    if (pathParts[pathParts.length - 2] === 'multipage') {
```
Comment (michaelficarra) [inline js/menu.js:954]: Is there a less fragile test? This appears to get tripped up on trailing slashes.

**Review by michaelficarra — APPROVED** [review]

File: js/menu.js (hunk 2 above)
Comment (bakkot) [reply js/menu.js:954]: What do you mean by "tripped up"? The only valid paths are `multipage/` and `multipage/some-file.html`, and

```js
let pathParts = 'https://tc39.es/ecma262/multipage/'.split('/');
pathParts[pathParts.length - 2] === 'multipage'; // true
```

```js
let pathParts = 'https://tc39.es/ecma262/multipage/index.html'.split('/');
pathParts[pathParts.length - 2] === 'multipage'; // true
```

`multipage` is a directory name, so the trailing slash is expected.

Comment (michaelficarra) [reply js/menu.js:954]: `multipage//`?

Comment (bakkot) [reply js/menu.js:954]: Not a valid url; see for yourself: https://tc39.es/ecma262/multipage//

File: js/menu.js (hunk 1 above)
Comment (bakkot) [reply js/menu.js]: Done.

**Pull request merged**